src/test/java/com/revature/studyforce/flashcard/integration/VoteIntegrationTest.java

```diff
@@ -0,0 +1,162 @@
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+/**
+ * Test class for VoteController {@link VoteController}
+ * @author Nick Zimmerman
+ */
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+@SpringBootTest
+@AutoConfigureMockMvc
+@TestPropertySource(locations = "classpath:test-application.properties")
+class VoteIntegrationTest {
+
+    @Autowired
+    private VoteRepository voteRepository;
+    @Autowired
+    private UserRepository userRepository;
+    @Autowired
+    private AnswerRepository answerRepository;
+    @Autowired
+    private VoteService voteService;
+    @Autowired
+    private FlashcardRepository flashcardRepository;
+    @Autowired
+    private VoteController controller;
+
+
+    @Test
+    void givenVoteDTO_whenCreateAnswerVote_shouldReturnVote() throws Exception {
+        User u = new User(0,"jesus.christ@revature.com","password","Jesus","Christ",true,false,false, Authority.USER, Timestamp.valueOf(LocalDateTime.now()),Timestamp.valueOf(LocalDateTime.now()));
+        userRepository.save(u);
+
+        Flashcard flashcard = new Flashcard(0,u,null,"how is your day",1,1,null,null,false);
+        flashcardRepository.save(flashcard);
+
+        Answer a = new Answer(0,u,flashcard,"check stackoverflow",5,false,false,Timestamp.valueOf(LocalDateTime.now()),Timestamp.valueOf(LocalDateTime.now()));
+        answerRepository.save(a);
+
+        MockMvc mockMvc = MockMvcBuilders.standaloneSetup(controller).build();
+        String string = "{\"answerId\" : \"3\","
+                + "\"userId\" : \"1\","
+                + "\"value\" : \"1\""
+                + "}";
+        System.out.println(string);
+        MvcResult result = mockMvc.perform(MockMvcRequestBuilders.post("/flashcards/votes/")
+                .content(string)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        System.out.println(result.getResponse().getContentAsString());
+        System.out.println(result.getResponse().getStatus());
+    }
+
+    @Test
+    void givenVoteWithBadValue_whenCreateAnswerVote_shouldThrowException() throws Exception {
+        User u = new User(0,"jesus.christ@revature.com","password","Jesus","Christ",true,false,false, Authority.USER, Timestamp.valueOf(LocalDateTime.now()),Timestamp.valueOf(LocalDateTime.now()));
```

Comment:
Using outdated user model, please correct to user model with 1 name field and no password field